A newer version of code with all the errors mentioned in the last pull request fixed

D0048 opened the pull request:
Still, there's always a lot to learn for me. Please check if this version meet your standard. It's really helping.  : ) 

Say, I am trying to add a new option to log all the wrong attempts somewhere. But I am not sure which solution is the best. Here's the solutions:
1.Log to a file
2.Display the log through frame-buffer
3.Both the above
I am not sure which one is better or is there a better solution. What do you think?

D0048 committed:
Reformat code|add code for the custom password feature|add help function

D0048 committed:
Add descrptions of how to use custom password into manual

D0048 committed:
Use getopt|Add debug macro|Add check for Nullptrs|reformatted code

All kinds of fixing of the errors pointed out by the last pull request

D0048 committed:
Add debug task to Makefile for debug builds

D0048 committed:
Add warning to one letter passwords

jaseg commented on xtrlock.c:
```diff
@@ -166,48 +205,103 @@ int main(int argc, char **argv){
+                                fprintf(stderr,"strdup:%s\n", strerror(errno)); 
+                                cust_pw_setting.pwd = strdup(crypt(optarg, optarg));/*never freed, fine in this case*/
+                            }
+                    }
```
Nooooooooooo!!!1!

Brief primer: If ```strdup``` fails and returns ```NULL``` that means you're out of memory. That condition can persist for several seconds or even minutes. Dumbly retrying a certain number of times does not work!

If ```strdup``` returns ```NULL```, the only sane way of action is to *abort operation right now with a visible error message*. Now, that would be a problem if the screen was already locked, but in this case it is perfectly fine as the screen locker is only about to start.

It may make sense to flash the screen briefly on successful startup to give the user an easily visible notification that her screen has *indeed* been locked.

BTW, what do you mean with the "never freed[...]" comment?

D0048 replied:
Good point. I will fix that in my next version! For the "never freed[...]" comment, I mean that piece of memory is on heap and is never freed by `free`. Just reminding myself that it is a possible source to memory leak.

jaseg replied:
That's good thinking.

D0048 replied:
Seems like I have some problem with xorg APIs flashing the screen. I may need some more time with it. However, I got all the other problems mentioned previously fixed. Please have a look. Thanks. : )

jaseg commented on xtrlock.man:
```diff
@@ -48,3 +48,18 @@ This program is distributed in the hope that it will be useful, but
+that one may not want to reveal. Using another password and share it 
+with your friend could solved this. However, due to the reason that 
+using an one character password may cause error, it is not allowed to
+use it as a custom password.
```
What is the problem with single-character passwords (apart from them being super insecure)? I think I don't understand yet.

jaseg commented on xtrlock.c:
```diff
@@ -166,48 +205,103 @@ int main(int argc, char **argv){
+    bool need_lock = false;
+        /*area for any arg init*/
+    if(!init_cust_pw
+    ){  fprintf(stderr,"Failed to init custom password config");
```
The ```){``` is on the wrong line here.

jaseg commented on xtrlock.c:
```diff
@@ -166,48 +205,103 @@ int main(int argc, char **argv){
+        
+        }
+    if(need_lock)
+            lock();
```
indentation

jaseg commented:
Apart from the one wtf, looks good to me :)

D0048 committed:
Add task: clean

D0048 committed:
Problem fixed: See details in description

1. Solve segementation fault on one-character passwords.
2. Remove the wtf error handling.
3. No longer require root access when using custom password
4. Add entry for indication of a successful lock(not implemented yet)

D0048 committed:
Remove contents disallowing single-character passwords

D0048 committed:
A missing line break found and added

D0048 committed:
Blink screen on successful lock|Fix argument handling bugs|+block screen

Now certain flags no longer require an arguments

D0048 committed:
Add descrption of block screen && blink screen functions

D0048 commented:
Well... I guess all the required changes have been applied. I also recover the function to completely block the screen. Please check them out in my newer commits : )
